docs/examples/src/main.cpp

```diff
@@ -0,0 +1,27 @@
+#include "divide.hpp"
+
+#include <iostream>
+#include <stdlib.h>
+#include <vector>
+
```

Comment:
I suggest the following simplified example

void my_print(int *i){
  printf("My integer:%d",*i);
}

int main(int argc, char* argv[]){
  int *undef_var;
  long tc= strtol(argv[1], nullptr, 10);
 
  switch(argc){
      case 0:
          printf("2nd argument:%s",argv[1]);//dereferencing an unitialized string
          break;
      case 1:
          divide(1,1-tc);//division by zero, only detected when ctu analysis is on
         break;
     case 2:
        //in case case0 would be detected by clangsa 
        my_print(undef_var); //dereferencing and unitialized pointer in a local function
        break;
  }
return 0;
}

}

Reply:
Not the same, but done.